libs/designsystem/calendar/src/calendar.component.ts

```diff
@@ -434,10 +435,15 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
     }
   }
 
-  _onDateSelected(newDay: CalendarCell) {
-    if (newDay.isSelectable && newDay.date) {
-      let newDate = new Date(this.activeMonth);
-      newDate.setDate(newDay.date);
+  _onDateSelected({ year, monthIndex, date, isSelectable }: CalendarCell) {
+    if (!isSelectable) return;
+
+    const isValidDate = date >= 1 && date <= 31;
+    const isValidMonth = monthIndex >= 0 && monthIndex <= 11;
+    const isValidYear = Number.isInteger(year);
+
+    if (isValidDate && isValidMonth && isValidYear) {
+      let newDate = new Date(year, monthIndex, date);
```

Comment:
Consider readability - as I recall we don't usually destructure function parameters? 🤔 
```suggestion
  _onDateSelected(newDay: CalendarCell) {
    if (!newDay.isSelectable) return;

    const isValidDate = newDay.date >= 1 && newDay.date <= 31;
    const isValidMonth = newDay.monthIndex >= 0 && newDay.monthIndex <= 11;
    const isValidYear = Number.isInteger(newDay.year);

    if (isValidDate && isValidMonth && isValidYear) {
      let newDate = new Date(newDay.year, newDay.monthIndex, newDay.date);
```